R/readGWLdata2.R

```diff
@@ -0,0 +1,120 @@
+# Copyright 2015 Province of British Columbia
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+# http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+#'
+#' Read in groundwater data from files downloaded from GWL tool
+#' @param path The path to the csv with historical data
+#' @param path The path to the csv with current data
+#' @param stnName The station name to use as it is no longer in data
+#' @param emsID The EMS ID of the well retained for compatibility
+#' @export
+#' @return A dataframe of the groundwater level observations
+#' @examples \dontrun{
+#'
+#'}
+readGWLdata2 <- function(historical, datafile, stnName="Not Available", emsID="NA") {
+  
+  if (!inherits(historical, "textConnection") && !file.exists(historical)) {
+    stop(paste0("The historical file ", historical, " does not exist."))
+  }
+
+  if (!inherits(datafile, "textConnection") && !file.exists(datafile)) {
+    stop(paste0("The data file ", datafile, " does not exist."))
+  }
+  
+  # get the well ID from the filenames and try to check they match
+  histfilename <- strsplit(historical, "[.]")[[1]][-2]
+  histwellid <- substr(histfilename, 3, 5)
+  
+  filename <- strsplit(datafile, "[.]")[[1]][-2]
+  wellid <- substr(filename, 3, 5)
+  
+  # Just warn, don't stop, in case user changed the file names
+  if (!identical(histwellid, wellid)) {
+    warning("The Well ID in file names do not match. Are the files both for the same well?")
+  }
+  
+  # load hisorical data
+  historicaldf <- read.csv(historical, row.names=NULL, stringsAsFactors=TRUE)
+  
+  # check column count, historical file should be 6: 
+  # dummydate, Value, type, year, month, daynum
+  if (length(names(historicaldf)) != 6) {
+    stop(paste0("Invalid number of fields in historical file. Expecting 6 and see ", length(names(historicaldf))))
+  }
+  
+  # put measures into columns using reshape2 library
+  library(reshape2)
```

Comment:
Instead of loading this with a `library` call, can you add it to the roxygen header with:
`'#  @importFrom reshape2 dcast`. That way it will get added as an import in the NAMESPACE